File: 2021/0001_twoSum/210622/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashTable = dict()
        for i, n in enumerate(nums):
            if target - n in hashTable:
                return [hashTable[target - n], i]
            hashTable[n] = i
        return []

File: 2021/0001_twoSum/210622/twoSum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::collections::HashMap;

        let len = nums.len();
        let mut map = HashMap::with_capacity(len);
        for i in 0..len {
            if let Some(k) = map.get(&(target - nums[i])) {
                return vec![*k as i32, i as i32];
            }
            map.insert(nums[i], i);
        }
        vec![]
    }
}
